Add AKS platform support (#219)
kinvolk/lokomotive: docs/configuration-reference/platforms/aks.md

# Lokomotive AKS configuration reference

## Contents

* [Introduction](#introduction)
* [Prerequisites](#prerequisites)
* [Configuration](#configuration)
* [Attribute reference](#attribute-reference)
* [Applying](#applying)
* [Destroying](#destroying)

## Introduction

This configuration reference provides information on configuring a Lokomotive cluster on Azure AKS with all the configuration options available to the user.

## Prerequisites

> * `lokoctl` [installed locally.](../../installer/lokoctl.md)

Reviewer: Nit
```suggestion
* `lokoctl` [installed locally](../../installer/lokoctl.md).
```

  client_id       = var.client_id
  client_secret   = var.client_secret

  location              = var.location
  resource_group_name   = var.resource_group_name
  application_name      = var.application_name
  manage_resource_group = var.manage_resource_group

  worker_pool "default" {
    count   = var.workers_count
    vm_size = var.workers_vm_size

    labels = {
      "key" = "value",
    }

    taints = [
      "node-role.kubernetes.io/master=NoSchedule",
    ]
  }

  tags = {
    "key" = "value",
  }
}
```

**NOTE**: Should you feel differently about the default values, you can set default values using the `variable`
block in the cluster configuration.

## Attribute reference

| Argument                | Description                                                  |    Default    | Required |
| ----------------------- | ------------------------------------------------------------ | :-----------: | :------: |
| `asset_dir`             | Location where Lokomotive stores cluster assets.             |       -       |   true   |
| `cluster_name`          | Name of the cluster. **NOTE**: It must be unique per resource group. |       -       |   true   |

> | `tenant_id`             | Azure Tenant ID. Can be also provided using `LOKOMOTIVE_AKS_TENANT_ID` environment variable. |       -       |   true   |
> | `subscription_id`       | Azure Subscription ID. Can be also provided using `LOKOMOTIVE_AKS_SUBSCRIPTION_ID` environment variable. |       -       |   true   |

Reviewer: ```suggestion
| `tenant_id`             | Azure Tenant ID. Can also be provided using the `LOKOMOTIVE_AKS_TENANT_ID` environment variable. |       -       |   true   |
| `subscription_id`       | Azure Subscription ID. Can also be provided using the `LOKOMOTIVE_AKS_SUBSCRIPTION_ID` environment variable. |       -       |   true   |
```

> | `resource_group_name`   | Name of resource group, where AKS cluster object will be created. Please note, that AKS will create also separate resource group for workers and other required objects, like load balancers, disks etc. If `manage_resource_group` parameter is set to `false`, this resource group must be manually created before cluster creation. |       -       |   true   |

Reviewer: ```suggestion
| `resource_group_name`   | Name of the resource group where AKS cluster object will be created. Please note, that AKS will also create a separate resource group for workers and other required objects, like load balancers, disks etc. If `manage_resource_group` parameter is set to `false`, this resource group must be manually created before cluster creation. |       -       |   true   |
```

> | `client_id`             | Azure service principal ID used  for running AKS cluster. Can be also provided using `LOKOMOTIVE_AKS_CLIENT_ID`. This parameter is mutually exclusive with `application_name` parameter. |       -       |  false   |

Reviewer: ```suggestion
| `client_id`             | Azure service principal ID used for running the AKS cluster. Can also be provided using `LOKOMOTIVE_AKS_CLIENT_ID`. This parameter is mutually exclusive with `application_name`. |       -       |  false   |
```

> | `client_secret`         | Azure service principal secret used  for running AKS cluster. Can be also provided using `LOKOMOTIVE_AKS_CLIENT_SECRET`. This parameter is mutually exclusive with `application_name` parameter. |       -       |  false   |

Reviewer: ```suggestion
| `client_secret`         | Azure service principal secret used for running the AKS cluster. Can also be provided using `LOKOMOTIVE_AKS_CLIENT_SECRET`. This parameter is mutually exclusive with `application_name`. |       -       |  false   |
```

| `tags`                  | Additional tags for Azure resources.                  |       -       |  false   |
| `location`              | Azure location where resources will be created. Valid values can be obtained using the following command from Azure CLI: `az account list-locations -o table`. | "West Europe" |  false   |

> | `application_name`      | Azure AD application name. If specified, new Application will be created in Azure AD together with service principle, which will be used to run AKS cluster on behalf of the user to provide full automation of cluster creation. Please note, that this require [permissions to create applications in Azure AD](https://docs.microsoft.com/en-us/azure/active-directory/users-groups-roles/roles-delegate-app-roles). This parameter is mutually exclusive with `client_id` and `client_secret` parameters. |       -       |  false   |

Reviewer: ```suggestion
| `application_name`      | Azure AD application name. If specified, a new Application will be created in Azure AD together with a service principal, which will be used to run the AKS cluster on behalf of the user to provide full cluster creation automation. Please note that this requires [permissions to create applications in Azure AD](https://docs.microsoft.com/en-us/azure/active-directory/users-groups-roles/roles-delegate-app-roles). This parameter is mutually exclusive with `client_id` and `client_secret`. |       -       |  false   |
```

> | `manage_resource_group` | If `true`, resource group for AKS object will be created on behalf of the user. |     true      |  false   |

Reviewer: ```suggestion
| `manage_resource_group` | If `true`, a resource group for the AKS object will be created on behalf of the user. |     true      |  false   |
```
